src/devices/Mhz19b/Mhz19b.cs

```diff
@@ -0,0 +1,251 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.IO.Ports;
+using System.Text;
+using System.Threading;
+using UnitsNet;
+
+namespace Iot.Device.Mhz19b
+{
+    /// <summary>
+    /// MH-Z19B CO2 concentration sensor binding
+    /// </summary>
+    public sealed class Mhz19b : IDisposable
+    {
+        private enum Command : byte
+        {
+            ReadCo2Concentration = 0x86,
+            CalibrateZeroPoint = 0x87,
+            CalibrateSpanPoint = 0x88,
+            AutoCalibrationSwitch = 0x79,
+            DetectionRangeSetting = 0x99
+        }
+
+        private enum MessageFormat
+        {
+            Start = 0x00,
+            SensorNum = 0x01,
+            Command = 0x02,
+            DataHighRequest = 0x03,
+            DataLowRequest = 0x04,
+            DataHighResponse = 0x02,
+            DataLowResponse = 0x03,
+            Checksum = 0x08
+        }
+
+        private const int MessageSize = 9;
+
+        private SerialPort _serialPort = null;
+        private Stream _serialPortStream = null;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Mhz19b"/> class using an existing (serial port) stream.
+        /// </summary>
+        /// <param name="stream">Existing stream</param>
+        public Mhz19b(Stream stream)
```

Comment:
Thanks for adding the stream. Just something that pops up in my mind when I was looking at the Dispose function: 
- should the stream just be disposed like you do now?
- should we use the same behavior as for GpioController, so adding a shouldDispose?
I'm more in favor of the second option. setting to null a stream that may be open thru a device than may have multiple streams or depends of some other elements seems a bit dangerous for me.

Reply:
I've to admit that I'm not that deep in the details of streams and their optimal behavior on disposal.
However, I don't mind having a look at the GpioController and try to adapt my implementation accordingly.

Reply:
then, just do something like that:
```csharp
private bool _shouldDispose:

public Mhz19b(Stream stream, bool shouldDispose = true)
{
  _shouldDispose = shouldDispose;
  // rest of your code
}

public Dispose()
{
  if (_shouldDispose)
  {
  }
}
```

Reply:
Done and pushed.